src/components/radio-group/radio-group.tsx

```diff
@@ -0,0 +1,196 @@
+                this.checkedArray.push(observable({checked: flag ? this.props.children[i].props.checked: false}));
+                if (this.props.children[i].props.checked) {
+                    flag = false;
+                }
+            }
+        } else if(this.props.dataSource) {
+            for (let i = 0; i < this.props.dataSource.length; i++) {
+                this.checkedArray.push(observable({checked: false}));
+            }
+        }
+        this.name = this.props.name ? this.props.name : 'name_' + counter++;
+    }
+
+    childrenOnClick(index: number) {
+        return (e: string) => {
+            this.checkedArray.forEach((value) => {
+                value.checked = false;
+            });
+            this.checkedArray[index].checked = true;
+            if (this.props.onChange) {
+                this.props.onChange(e);
+            }
+        };
+    }
+
+    createChildren(dataArray: any, isDataSource: boolean) {
+        const childArray: React.ReactNode[] = [];
+        for (let index = 0; index < dataArray.length; index++) {
+            const data = dataArray[index];
+            const props: RadioButtonProps = {
+                key: index,
+                value: '',
+                automationId: 'RADIO_BUTTON_' + index,
+                checked: this.checkedArray[index].checked,
+                onClick: this.childrenOnClick(index),
+                disabled: this.props.disabled,
+                location: this.props.location,
+                name: this.name};
+
+            if (isDataSource) {
+                props.value = data;
+                childArray.push(React.createElement(RadioButton, props));
+            } else {
+                if (data.type === RadioButton) {
+                    if ((data.props as RadioButtonProps).disabled) {
+                        props.disabled = true;
+                    }
+                    props.value = (data.props as RadioButtonProps).value;
+                    childArray.push(React.cloneElement(data as ReactElement<any>, props));
+                } else {
+                    childArray.push(React.cloneElement(data as ReactElement<any>, {key: index,
+                        checked: this.checkedArray[index].checked,
+                        onClick: action(this.childrenOnClick(index))}));
+                }
+            }
+        }
+        return childArray;
+    }
+
+    render() {
+        const dataArray = this.props.children ? this.props.children : this.props.dataSource;
+        let childArray: React.ReactNode[] = [];
+
+        if (this.props.dataSource!.length > 0) {
+            childArray = this.createChildren(this.props.dataSource, true);
+        } else if (React.isValidElement(dataArray)) {
+            childArray.push(dataArray);
+        } else {
+            if (this.props.children) {
+                childArray = this.createChildren(this.props.children, false);
+            }
+        }
+
```

Comment:
to push the point made by @alisey , It would be better if we had a clear logic to prefer dataSource vs. children, and then maybe have two separate methods to "pack" it to childArray. something like ```childArray = this.props.children ? populateFromChildren(this.props.children) : populateFromDataSource(this.props.dataSource)```